### PR TITLE
109.event reason word limit

### LINKED ISSUE #109: Limit Event Reason to 3 Words
[//]: # (These are comments and are used for reference and will not show up in the PR)

User Requirement: 02.01.01

[//]: # (Please Add labels for type of issue)

Description: This is a defect that been brought up to our attention that we limit the number of characters to 20 for a textual reason for a vibe event but do not consider limiting the word count to 3. 

I found [this](https://stackoverflow.com/questions/11839094/edittext-input-limit-500-words) as a potential resource for implementing a TextWatcher so we can do exactly that. 

[//]: # (If an issue is a bug or defect, please list steps to be able to reproduce bug and a screenshot of the defect i.e terminal message)

Story Points: 2

Related Issues: #79 (already closed)

Task List

- [ ] Decide how to proceed with limiting word count (pls let team know)
- [ ] Defect resolved i.e. vibe event reason is limited to 3 words
- [ ] Unit test if necessary
- [ ] Intent/UI test if necessary
- [ ] Properly commented functionality so Javadocs can be generated

[//]: # (Notes:)
[//]: # (- Testing and Documentation tasks should always be assumed, but only need to be explicitly stated if there is something that ~~does not~~ need modification. For example,)
[//]: # (some Issues/Stories may not require one of Unit or Intent tests, and this should be indicated in the task list.)
[//]: # (- Code Reviews are also always assumed to be required, but an issue should only indicate in the description if a review is not required, but someone still shouldn't)
[//]: # (merge their own pull request)

### PR DESCRIPTION
**Pull request by TheBroMoe**
[//]: # (These are comments and are used for reference and will not show up in the PR)

Description: Fix bug where user may enter more than 3 words in reason editText field

[//]: # (Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change)

[//]: # (automatically closes given issue number)
Closes #109

Proof of Concept

[//]: # (Attach Image of feature added or fix applied and working)
[//]: # (Please provide images of tests passing)
itwork

Checklist

[//]: # (Add an x between square brackets to mark item as done)

- [x] Follows style guidelines and is **properly commented**
- [x] Code changes generate no new warnings
- [x] Testing (Unit and/or Intent)
- [x] Documentation (UML, Wiki, etc.)

### CODE REVIEW

File: app/src/main/java/com/example/vybe/AddEditVibeActivity.java [hunk 1]
```diff
@@ -92,6 +93,7 @@ protected void onCreate(Bundle savedInstanceState) {
 
         imageView.setDrawingCacheEnabled(true);
         imageView.buildDrawingCache();
+        reasonMaxWords = 3;
```
Comment (raysarinas) [inline app/src/main/java/com/example/vybe/AddEditVibeActivity.java]: move this to util->Constants.java file

File: app/src/main/java/com/example/vybe/AddEditVibeActivity.java [hunk 2]
```diff
@@ -161,9 +163,15 @@ public void onNothingSelected(AdapterView<?> adapterView) {
...
 
             vibeEvent.setDateTime(LocalDateTime.of(selectedDate, selectedTime));
-            vibeEvent.setReason(reasonField.getText().toString());
-
 
+            if (reasonField.getText().toString().trim().split("\\s").length <= reasonMaxWords) {
+                reasonField.setError(null);
+                vibeEvent.setReason(reasonField.getText().toString());
+            } else {
+                // your code here
```
Comment (raysarinas) [inline app/src/main/java/com/example/vybe/AddEditVibeActivity.java]: ?

File: app/src/main/java/com/example/vybe/AddEditVibeActivity.java [hunk 3]
```diff
@@ -161,9 +163,15 @@ public void onNothingSelected(AdapterView<?> adapterView) {
...
+                reasonField.setError(String.format("max %d words allowed", reasonMaxWords));
```
Comment (raysarinas) [inline app/src/main/java/com/example/vybe/AddEditVibeActivity.java]: Capitalize first word "max" please

File: app/src/main/java/com/example/vybe/AddEditVibeActivity.java [hunk 4]
```diff
@@ -161,9 +163,15 @@ public void onNothingSelected(AdapterView<?> adapterView) {
...
+                return;
```
Comment (raysarinas) [inline app/src/main/java/com/example/vybe/AddEditVibeActivity.java]: what?

File: app/src/main/java/com/example/vybe/AddEditVibeActivity.java (hunk 2 above)
Comment (TheBroMoe) [reply app/src/main/java/com/example/vybe/AddEditVibeActivity.java]: Comment removed

File: app/src/main/java/com/example/vybe/AddEditVibeActivity.java (hunk 4 above)
Comment (TheBroMoe) [reply app/src/main/java/com/example/vybe/AddEditVibeActivity.java]: return here is used to make sure app doesn't exit activity after clicking the save button and returns process to onCreate

**Comment by TheBroMoe** [comment]
Changes have been completed :)

**Pull request merged**